rdbms/src/main/java/org/spine3/server/storage/jdbc/entity/query/SelectBulkQuery.java

```diff
@@ -22,91 +22,145 @@
+ * <p>{@code SQL} analogs of this are:
+ * <ul>
+ *     <li>1. {@code SELECT * FROM table;}
+ *     <li>2. {@code SELECT * FROM table WHERE id IN (?,...,?);}
+ * <ul/>
+ *
  * @author Dmytro Dashenkov
  */
-public class SelectBulkQuery<M extends Message> extends Query {
+public class SelectBulkQuery extends StorageQuery {
 
-    private final Descriptors.Descriptor messageDescriptor;
     private final TypeUrl typeUrl;
     private final FieldMask fieldMask;
+    private final List arguments;
 
-    private static final String COMMON_TEMPLATE = "SELECT * FROM TABLE %s";
+    private static final String COMMON_TEMPLATE = "SELECT * FROM %s";
     private static final String ALL_TEMPLATE = COMMON_TEMPLATE + ';';
     @SuppressWarnings("DuplicateStringLiteralInspection")
     private static final String IDS_TEMPLATE = COMMON_TEMPLATE + " WHERE " + EntityTable.ID_COL + " IN (%s);";
 
     private static final int IDS_STRING_ESTIMATED_LENGTH = 128;
 
-    protected SelectBulkQuery(Builder<M> builder) {
+    protected SelectBulkQuery(Builder builder) {
         super(builder);
-        this.messageDescriptor = checkNotNull(builder.messageDescriptor);
-        this.fieldMask = builder.fieldMask;
+        final Descriptors.Descriptor messageDescriptor = checkNotNull(builder.messageDescriptor);
         this.typeUrl = TypeUrl.of(messageDescriptor);
+        this.fieldMask = builder.fieldMask;
+        this.arguments = builder.arguments;
     }
 
-    public Map<Object, M> execute() throws SQLException {
-        final ResultSet resultSet = SqlExecutionHelper.execute(getQuery(), getConnection(true));
+    /**
+     * Executes the query.
+     *
+     * @return ID-to-{@link EntityStorageRecord} {@link Map} as the result of the query.
+     * @throws SQLException if the input data contained SQL errors or the table does not exist.
+     */
+    public Map<Object, EntityStorageRecord> execute() throws SQLException {
+        final ConnectionWrapper connection = getConnection(true);
+        final PreparedStatement sqlStatement = connection.prepareStatement(getQuery());
 
-        return QueryResults.parse(resultSet, messageDescriptor, fieldMask, typeUrl);
+
+        for (int i = 0; i < arguments.size(); i++) {
+            sqlStatement.setObject(i + 1, arguments.get(i));
+        }
+
+        final ResultSet resultSet = sqlStatement.executeQuery();
+
+        connection.close();
+
+        return QueryResults.parse(resultSet, fieldMask, typeUrl);
     }
 
-    public static Builder<?> newBuilder(String tableName) {
+    /**
+     * @return New instance of the {@link Builder} set to "query all" by default.
+     */
+    public static Builder newBuilder(String tableName) {
         return newBuilder()
                 .setAllQuery(tableName);
     }
 
-    public static Builder<?> newBuilder() {
-        return new Builder<>();
+    /**
+     * @return New instance of the {@link Builder}.
```

Comment:
You don't use capitalization in case of `@return ...`. I.e. you could write `new instance ...`.